bank_statement_parse_mt940/mt940.py

```diff
@@ -0,0 +1,254 @@
+        for dummy_i in range(self.header_lines):
+            iterator.next()
+        self.current_statement = BankStatement()
+
+    def handle_footer(self, line, iterator):
+        """add current statement to list, reset state"""
+        self.statements.append(self.current_statement)
+        self.current_statement = None
+
+    def handle_record(self, line):
+        """find a function to handle the record represented by line"""
+        tag_match = re.match(self.tag_regex, line)
+        tag = tag_match.group(0).strip(':')
+        if not hasattr(self, 'handle_tag_%s' % tag):
+            logging.error('Unknown tag %s', tag)
+            logging.error(line)
+            return
+        handler = getattr(self, 'handle_tag_%s' % tag)
+        handler(line[tag_match.end():])
+
+    def handle_tag_20(self, data):
+        """Contains unique ? message ID"""
+        pass
+
+    def handle_tag_25(self, data):
+        """Handle tag 25: local bank account information."""
+        data = data.replace('EUR', '').replace('.', '').strip()
+        self.current_statement.local_account = data
+
+    def handle_tag_28C(self, data):
+        """Sequence number within batch - normally only zeroes."""
+        pass
+
+    def handle_tag_60F(self, data):
+        """get start balance and currency"""
+        # For the moment only first 60F record
+        # The alternative would be to split the file and start a new
+        # statement for each 20: tag encountered.
+        stmt = self.current_statement
+        if not stmt.local_currency:
+            stmt.local_currency = data[7:10]
+            stmt.start_balance = str2amount(data[0], data[10:])
+
+    def handle_tag_61(self, data):
+        """get transaction values"""
+        transaction = self.current_statement.create_transaction()
+        self.current_transaction = transaction
+        transaction.execution_date = datetime.strptime(data[:6], '%y%m%d')
+        transaction.value_date = datetime.strptime(data[:6], '%y%m%d')
+        #  ...and the rest already is highly bank dependent
+
+    def handle_tag_62F(self, data):
+        """Get ending balance, statement date and id.
+
+        We use the date on the last 62F tag as statement date, as the date
+        on the 60F record (previous end balance) might contain a date in
+        a previous period.
+
+        We generate the statement.id from the local_account and the end-date,
+        this should normally be unique, provided there is a maximum of
+        one statement per day.
+
+        Depending on the bank, there might be multiple 62F tags in the import
+        file. The last one counts.
+        """
+        stmt = self.current_statement
+        stmt.end_balance = str2amount(data[0], data[10:])
+        stmt.date = datetime.strptime(data[1:7], '%y%m%d')
+        stmt.statement_id = '%s-%s' % (
+            stmt.local_account,
+            stmt.date.strftime('%Y-%m-%d'),
+        )
```

Comment:
In the examples I have, I get a statement number in tag 28C. I thought preserving it in statement_id, but this part overwrites it. Perhaps this should be done only if statement_id has not been set during parsing?